sites/all/modules/reol_statistics/reol_statistics.module

```diff
@@ -51,24 +36,32 @@ function reol_statistics_publizon_loan($account, $name, $isbn, $retailer_id) {
 }
 
 /**
- * Implements hook_ding_unilogin_login().
+ * Get UNI•Login log data for the current user.
  *
- * Store statistics details in session for later logging.
+ * @return array
+ *   Array of collected data or an empty array.
  */
-function reol_statistics_ding_unilogin_login($user) {
-  $unilogin_data = _reol_statistics_get_unilogin_data($user->getName());
-  if (empty($unilogin_data)) {
-    watchdog(
-      'reol_statistics',
-      'UNI•Login failure, denying login for %user.',
-      array('%user' => $user->getName()),
-      WATCHDOG_ERROR
-    );
-    throw new Exception('Error communicating with UNI•Login');
+function _reol_statistics_get_unilogin_data() {
+  global $user;
+
+  if (!isset($user->data['unilogin']['institutions'])
```

Comment:
How is the user logged in and can the use get the same drupal user object logging in with different login methods CPR vs. unilog vs. adgangsplatformen

Reply:
If the same drupal user object gets load this may give wrong stats.

Reply:
Unless I'm completely missing missing something this should never happen (cf. https://github.com/eReolen/ereolengo/blob/develop/ding_unilogin/ding_unilogin.module#L444-L447) – but it may be better to store this data in the session.

@cableman, are we sure that the session (i.e. `$_SESSION`) is cleared on logout?

Reply:
User data now moved into `$_SESSION`.